browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -281,6 +282,10 @@ void SpeedreaderTabHelper::ProcessNavigation(
 
   auto* rewriter_service =
       g_brave_browser_process->speedreader_rewriter_service();
+  if (!rewriter_service) {
```

Comment:
this already has `MaybeCreateForWebContents` so we should check `g_brave_browser_process->speedreader_rewriter_service` and then pass it into the constructor using 

```
web_contents->SetUserData(
        kWebContentsUserDataKey,
        std::make_unique<SpeedReaderTabHelper>(
            web_contents, rewriter_service));
```

Reply:
Done in [a3ec810](https://github.com/brave/brave-core/pull/20433/commits/a3ec810d756633f63eae8ab4c0f938e30fcaa386)